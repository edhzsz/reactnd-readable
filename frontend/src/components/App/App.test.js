import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import ConnectedApp, { mapStateToProps, App } from './App';

Enzyme.configure({ adapter: new Adapter() });

describe('Component: App', () => {

  describe('Map Dispatch To Props', () => {
    it('returns the default state', () => {
      expect(mapStateToProps()).toEqual({
        name: 'Edgar'
      });
    });
  });

  describe('Disconnected component', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<App />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('should match its empty snapshot', () => {
      const tree = shallow(
        <App />
      );

      expect(toJSON(tree)).toMatchSnapshot();
    });

    it('should render the name prop', () => {
      const tree = shallow(
        <App name='Test'/>
      );

      expect(toJSON(tree)).toMatchSnapshot();
    });
  });

  describe('Connected component', () => {
    it('should be connected', () => {
      const mockStore = configureMockStore();
      const store = mockStore({

      });

      const tree = renderer.create(
        <Provider store={store} >
          <ConnectedApp />
        </Provider>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});