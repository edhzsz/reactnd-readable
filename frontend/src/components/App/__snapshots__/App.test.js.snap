// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App Connected component should be connected 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    Hello 
    Edgar
    !
  </p>
</div>
`;

exports[`Component: App Disconnected component should match its empty snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    Hello 
    !
  </p>
</div>
`;

exports[`Component: App Disconnected component should render the name prop 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    Hello 
    Test
    !
  </p>
</div>
`;
